Content.Shared/Toilet/SharedToiletSystem.cs

```diff
@@ -0,0 +1,182 @@
+        [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+        [Dependency] private readonly SharedAudioSystem _audio = default!;
+        [Dependency] private readonly SecretStashSystem _secretStash = default!;
+        [Dependency] private readonly SharedToolSystem _tool = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<ToiletComponent, ComponentStartup>(OnComponentStartup);
+            SubscribeLocalEvent<ToiletComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<ToiletComponent, GetVerbsEvent<AlternativeVerb>>(OnToggleSeatVerb);
+            SubscribeLocalEvent<ToiletComponent, ActivateInWorldEvent>(OnActivateInWorld);
+            SubscribeLocalEvent<ToiletComponent, ToiletPryDoAfterEvent>(OnToiletPried);
+            SubscribeLocalEvent<ToiletComponent, InteractUsingEvent>(OnInteractUsing);
+            SubscribeLocalEvent<ToiletComponent, InteractHandEvent>(OnInteractHand);
+            SubscribeLocalEvent<ToiletComponent, ExaminedEvent>(OnExamine);
+        }
+
+        private void OnInit(EntityUid uid, ToiletComponent component, ComponentInit args)
+        {
+            EnsureComp<SecretStashComponent>(uid);
+        }
+        private void OnComponentStartup(EntityUid uid, ToiletComponent component, ComponentStartup args)
+        {
+            UpdateAppearance(uid, component);
+        }
+
+        private void OnInteractUsing(EntityUid uid, ToiletComponent component, InteractUsingEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            // are player trying place or lift of cistern lid?
+            else if (_tool.UseTool(args.Used, args.User, uid, component.PryLidTime, component.PryingQuality, new ToiletPryDoAfterEvent()))
+            {
+                args.Handled = true;
+            }
+            // maybe player trying to hide something inside cistern?
+            else if (component.ToggleLid)
+            {
+                _secretStash.TryHideItem(uid, args.User, args.Used);
+                args.Handled = true;
+            }
+        }
+
+        private void OnInteractHand(EntityUid uid, ToiletComponent component, InteractHandEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            // trying get something from stash?
+            if (component.ToggleLid)
+            {
+                var gotItem = _secretStash.TryGetItem(uid, args.User);
+                if (gotItem)
+                {
+                    args.Handled = true;
+                    return;
+                }
+            }
+
+            args.Handled = true;
+        }
+
+        private void OnToggleSeatVerb(EntityUid uid, ToiletComponent component, GetVerbsEvent<AlternativeVerb> args)
+        {
+            if (!args.CanInteract || !args.CanAccess || !CanToggle(uid) || !HasComp<HandsComponent>(args.User))
```

Comment:
```suggestion
            if (!args.CanInteract || !args.CanAccess || !CanToggle(uid) || args.Hands == null)
```

gets passed from `GetVerbsEvent` already